cachi2/core/package_managers/rpm/main.py

```diff
@@ -0,0 +1,204 @@
+        # NOTE: add license into SBOM?
+        rpm_args = [
+            "-q",
+            "--queryformat",
+            "%{LICENSE}",
+            file_path,
+        ]
+        license = run_cmd(cmd=["rpm", *rpm_args], params={})
+
+        with rpmfile.open(file_path) as rpm:
+            vendor = rpm.headers.get("vendor", b"").decode()
+            vendor = quote(vendor.lower())
+
+            name = rpm.headers.get("name", b"").decode()
+            version = rpm.headers.get("version", b"").decode()
+            release = rpm.headers.get("release", b"").decode()
+            arch = rpm.headers.get("arch", b"").decode()
+            download_url = quote(file_metadata["url"])
+
+            # https://github.com/package-url/purl-spec/blob/master/PURL-TYPES.rst#rpm
+            # https://github.com/package-url/purl-spec/blob/master/PURL-SPECIFICATION.rst#known-qualifiers-keyvalue-pairsa
+            purl = (
+                f"pkg:rpm{'/' if vendor else ''}{vendor}/{name}@{version}{release}"
+                f"?arch={arch}{'&epoch=' if epoch else ''}{epoch}&download_url={download_url}"
+            )
+
+            components.append(
+                Component(
+                    name=name,
+                    version=version,
+                    purl=purl,
+                    # NOTE: add checksum into SBOM?
+                )
+            )
+
+
+def process_packages(path: Path) -> None:
+    """
+    Search structure for all repoid dirs and create repository metadata
+    out of its RPMs. Skip sources dir that contains only SRPMs.
+    """
+    package_dir = RootedPath(path).join_within_root(DEFAULT_PACKAGE_DIR)
+    for entry in os.listdir(package_dir):
+        if entry == "sources" or not os.path.isdir(os.path.join(package_dir, entry)):
+            continue
+        arch = entry
+        for entry in os.listdir(os.path.join(package_dir, arch)):
+            if not os.path.isdir(os.path.join(package_dir, arch, entry)):
+                continue
+            repoid = entry
+            repodir = os.path.join(package_dir, arch, repoid)
+            createrepo(repoid, repodir)
+
+
+def createrepo(repoid: str, repodir: str) -> None:
+    log.info(f"Creating repository metadata for repoid '{repoid}': {repodir}")
+    cmd = ["/usr/bin/createrepo", repodir]
+    log.debug("$ " + shlex.join(cmd))
+    stdout = run_cmd(cmd, params={})
+    log.debug(stdout)
+
+
+def _generate_repofiles(path: RootedPath) -> list[ProjectFile]:
+    """
+    Search structure and generate repofile for each arch in its dir.
+    Repofile contains all arch's repoids. Skip sources dir that contains only SRPMs.
+    """
+    project_files: list[ProjectFile] = []
+    package_dir = path.join_within_root(DEFAULT_PACKAGE_DIR)
+    for subdir in os.listdir(package_dir):
+        if subdir == "sources":
+            continue
+        arch = subdir
+        log.debug(f"Preparing repofile for arch '{arch}'")
+        abspath = os.path.join(package_dir, arch, "cachi2.repo")
```

Comment:
Repo files should live separately from actual packages. IIUC the idea is to do something like `podman build -v <repos>:/etc/yum.repos.d ...` so I can imagine another subdirectory for the repos like `cachi2-output/deps/rpm/repos.d/cachi2.repo`

Reply:
I am making _cachi2.repo_ file for every architecture so I can't store it in a single directory. Do you think creating corresponding arch subdirectories only with a single .repo file would be OK?

Reply:
> I am making cachi2.repo file for every architecture so I can't store it in a single directory. Do you think creating corresponding arch subdirectories only with a single .repo file would be OK?

No, we already have the `<arch>` directory, so better nested it there like this: `cachi2-output/deps/rpm/<arch>/repos.d/cachi2.repo`